RecoBTag/TensorFlow/plugins/DeepDoubleBTagInfoProducer.cc

```diff
@@ -0,0 +1,350 @@
+            }
+        }
+
+        // sort collections (open the black-box if you please)
+        std::sort(c_sorted.begin(), c_sorted.end(),
+            btagbtvdeep::SortingClass<std::size_t>::compareByABCInv);
+
+        std::vector<size_t> c_sortedindices;
+
+        // this puts 0 everywhere and the right position in ind
+        c_sortedindices = btagbtvdeep::invertSortingVector(c_sorted);
+
+        // set right size to vectors
+        features.c_pf_features.clear();
+        features.c_pf_features.resize(c_sorted.size());
+
+        for (unsigned int i = 0; i < daughters.size(); i++)
+        {
+            auto const* cand = daughters.at(i);
+            if (cand)
+            {
+                // candidates under 950MeV are not considered
+                // might change if we use also white-listing
+                if (cand->pt() < 0.95)
+                    continue;
+
+                auto reco_cand = dynamic_cast<const reco::PFCandidate*>(cand);
+
+                // get PUPPI weight from value map
+                float puppiw = 1.0; // fallback value
+
+                float drminpfcandsv = btagbtvdeep::mindrsvpfcand(svs_unsorted, cand, jet_radius_);
+
+                if (cand->charge() != 0)
+                {
+                    // is charged candidate
+                    auto entry = c_sortedindices.at(i);
+                    // get cached track info
+                    auto& trackinfo = trackinfos.at(i);
+                    // get_ref to vector element
+                    auto& c_pf_features = features.c_pf_features.at(entry);
+                    // fill feature structure
+                    auto packed_cand = dynamic_cast<const pat::PackedCandidate*>(cand);
+                    if (packed_cand)
+                    {
+                        btagbtvdeep::packedCandidateToFeatures(
+                            packed_cand, jet, trackinfo, drminpfcandsv,
+                            static_cast<float>(jet_radius_), c_pf_features);
+                    }
+                    else if (reco_cand)
+                    {
+                        // get vertex association quality
+                        int pv_ass_quality = 0; // fallback value
+                        // getting the PV as PackedCandidatesProducer
+                        // but using not the slimmed but original vertices
+                        auto ctrack = reco_cand->bestTrack();
+                        int pvi = -1;
+                        float dist = 1e99;
+                        for (size_t ii = 0; ii < vtxs->size(); ii++)
+                        {
+                            float dz = (ctrack) ? std::abs(ctrack->dz(((*vtxs)[ii]).position())) : 0;
+                            if (dz < dist)
+                            {
+                                pvi = ii;
+                                dist = dz;
+                            }
+                        }
+                        auto PV = reco::VertexRef(vtxs, pvi);
```

Comment:
start local variables and data members with lower case letter. Capitalization is reserved for class names.